ARCHITECTURE.md

```diff
@@ -6,6 +6,35 @@ This document describes the grammar and production rules of Cucumber Expressions
 
 A Cucumber Expression has the following [EBNF](https://en.wikipedia.org/wiki/Extended_Backus%E2%80%93Naur_form) grammar:
 
+```ebnf
```

Comment:
Rather than adding a new EBNF grammar, can you please make modifications to the existing one? There should only be one grammar.

Reply:
Also, it's important that our implementations are consistent with the formal grammar, so if we're going to update it with your proposed changes, we may also need to change the existing implementations.

Reply:
@aslakhellesoy 

> can you please make modifications to the existing one

existing grammar is context-specific and can't be called EBNF at all. Also it describes superset of `Cucumber Expression` while there is no need to do it. 
Old specification if left to provide insights into current implementation.